src/core/thread/network_data.cpp

```diff
@@ -565,39 +565,27 @@ const PrefixTlv *NetworkData::FindPrefix(const uint8_t *aPrefix,
     return prefixTlv;
 }
 
-const ServiceTlv *NetworkData::FindService(uint32_t       aEnterpriseNumber,
-                                           const uint8_t *aServiceData,
-                                           uint8_t        aServiceDataLength) const
+const ServiceTlv *NetworkData::FindService(uint32_t             aEnterpriseNumber,
+                                           const uint8_t *      aServiceData,
+                                           uint8_t              aServiceDataLength,
+                                           ServiceDataMatchMode aServiceMatchMode) const
 {
-    return FindService(aEnterpriseNumber, aServiceData, aServiceDataLength, mTlvs, mLength);
+    return FindService(aEnterpriseNumber, aServiceData, aServiceDataLength, aServiceMatchMode, mTlvs, mLength);
 }
 
-const ServiceTlv *NetworkData::FindService(uint32_t       aEnterpriseNumber,
-                                           const uint8_t *aServiceData,
-                                           uint8_t        aServiceDataLength,
-                                           const uint8_t *aTlvs,
-                                           uint8_t        aTlvsLength)
-{
-    return FindService(aEnterpriseNumber, aServiceData, aServiceDataLength, /* aExactServiceDataMatch */ true, aTlvs,
-                       aTlvsLength);
-}
-
-const ServiceTlv *NetworkData::FindService(uint32_t       aEnterpriseNumber,
-                                           const uint8_t *aServiceData,
-                                           uint8_t        aServiceDataLength,
-                                           bool           aExactServiceDataMatch,
-                                           const uint8_t *aTlvs,
-                                           uint8_t        aTlvsLength)
+const ServiceTlv *NetworkData::FindService(uint32_t             aEnterpriseNumber,
+                                           const uint8_t *      aServiceData,
+                                           uint8_t              aServiceDataLength,
+                                           ServiceDataMatchMode aServiceMatchMode,
```

Comment:
```suggestion
                                           ServiceMatchMode aServiceMatchMode,
```

Or
`ServiceDataMatchMode aServiceDataMatchMode,`

Reply:
The type name 'ServiceMatchMode' would fit best, given the names of functions (FindService() , MatchService() )

Reply:
Thanks for the suggestion. Will fix.